### PR TITLE
Implement a full text search service for Items using SQLite fts5

### PR DESCRIPTION
**Pull request by brouberol**
The way this works is by defining an SQLite search index table and relying on triggers to keep the search index in sync with the actual `item` table.

Every time an `Item` is inserted to, deleted from, or updated in the `item` table, a trigger is executed to mirror the changes into the `item_search_index` virtual table.
This way, the search index is always up-to-date.

We also introduce a way to favor result of a given language, to make sure to surface the results in the player's locale/language first.

### CODE REVIEW

**Comment by brouberol** [comment]
Seems that the unit tests are failing due to a discrepancy between the SQLite version I run locally and the one running in CI.

**Comment by brouberol** [comment]
This implements part of #205, by we also need tu build the API on top of it.

Before I go any further, I'd need to make sure of the type of data we'd expect from the API (and ergo the repository) to display the search results. @etnbrd hinted to using https://github.com/thisbeyond/solid-select. I'll have a look at it and see what type of data they require

File: dnd5esheets/migrations/versions/623e8558c9ba_add_item_full_text_search_table.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+                SELECT
+                    cast(hex(json_each.key) as integer) + new.id,
+                    new.id,
+                    json_each.key,
+                    json_extract(json_each.value, '$.name'),
+                    json_extract(json_each.value, '$.description')
+                FROM json_each(json_extract(new.json_data, '$.meta.translations'));
+        END;
+        """,
+        # Populate the search index with the existing items
```
Comment (etnbrd) [inline dnd5esheets/migrations/versions/623e8558c9ba_add_item_full_text_search_table.py:52]: Is the order important between the triggers and the population of the table?

File: dnd5esheets/repositories/base.py [hunk 2]
```diff
@@ -46,3 +55,19 @@ async def delete_by_id(cls, session: AsyncSession, id: int):
...
+
+    @classmethod
+    async def _search(
+        cls, session: AsyncSession, search_term: str, sort_by_rank: bool = True
+    ) -> Sequence[Row[Any]]:
+        fields = ["rank"] + cls.search_fields
+        query = (
+            f"SELECT {', '.join(fields)} FROM {cls.search_index_table_name()} "
+            f"WHERE {cls.search_index_table_name()} MATCH :search_term "
+            f"{'ORDER BY rank' if sort_by_rank else ''};"
```
Comment (etnbrd) [inline dnd5esheets/repositories/base.py:67]: I don't see `rank` in the `item_search_index`. Is this something populated by fts5?

**Review by etnbrd — APPROVED** [review]
🙌

**Comment by etnbrd** [comment]
> Before I go any further, I'd need to make sure of the type of data we'd expect from the API (and ergo the repository) to display the search results.

It shouldn't depend on the client implementation. I guess we can just go with, as an input, a string that's a suite of words or just one word, and return as an output the list of results with their rank.

File: dnd5esheets/migrations/versions/623e8558c9ba_add_item_full_text_search_table.py (hunk 1 above)
Comment (brouberol) [reply dnd5esheets/migrations/versions/623e8558c9ba_add_item_full_text_search_table.py:52]: Not so much no, I could add the population of the table after the triggers. It's important that I define all of them, but the order of definition isn't important.

**Comment by brouberol** [comment]
Fixes #205

**Comment by brouberol** [comment]
```console
~/code/5esheets item-fts *1 ?1 ❯ curl -s "localhost:8000/api/item/search?search_term=name:chain&favored_language=en&limit=2" | jq .
[
  {
    "rank": -3.381911532797657,
    "resource_id": 8,
    "language": "en",
    "name": "Chain Mail"
  },
  {
    "rank": -2.837992364252012,
    "resource_id": 9,
    "language": "en",
    "name": "Chain Shirt"
  }
]
```

**Pull request merged**